packages/react-instantsearch/src/widgets/Hits.tsx

```diff
@@ -65,10 +82,19 @@ export function Hits<THit extends BaseHit = BaseHit>({
     </li>
   );
 
+  const banner = BannerComponent === false ? undefined : bannerData;
+
+  const bannerComponent =
+    BannerComponent === false
+      ? undefined
+      : (BannerComponent as HitsUiComponentProps<Hit<THit>>['bannerComponent']);
```

Comment:
I welcome a suggestion for better typing here to avoid the type assertion.

Reply:
I don't know if it's any better, but this works without casting:

```tsx
  const banner = BannerComponent ? bannerData : undefined;

  const bannerComponent = BannerComponent
    ? (bannerComponentProps: {
        banner: Required<HitsUiComponentProps<Hit<THit>>>['banner'];
        className: string;
      }) => <BannerComponent {...bannerComponentProps} />
    : undefined;
```

Reply:
Oh it's because in `react-instantsearch` we accept both function and class components through `React.JSXElementConstructor`, while in `instantsearch-ui-components` it's much more restrictive and only accepts function components that return a single `JSX.Element`.

I think what Taylor did is good and can be simplified with : 
```jsx
const bannerComponent = (
    BannerComponent === false ? () => null : BannerComponent
  ) as HitsUiComponentProps<Hit<THit>>['bannerComponent'];
```
That way we don't need to pass an undefined `banner`, we just pass a dummy component.

Later down the line we'll have to review `instantsearch-ui-component` types because I don't think instantiating component wrappers inline in render like this is good as it makes the user's component remount each time, which might raise issues later if we want to push single object ID hits.

Reply:
Thank you both. @aymeric-giraudet - in your opinion, what is the advantage of passing `() => null` instead of `undefined`? I'm just curious - I don't have any issue with making this change.